ratbag_emu/device.py

```diff
@@ -0,0 +1,272 @@
+from ratbag_emu.firmware import Firmware
+from ratbag_emu.hw_component import HWComponent
+from ratbag_emu.util import EventData, mm2inch, ms2s
+
+
+class Device(object):
+    '''
+    Represents a real device
+
+    :param name:    Device name
+    :param info:    Bus information (bus, vid, pid)
+    :param rdescs:  Array of report descriptors
+    '''
+    device_list: ClassVar[List[str]] = []
+
+    def __init__(self, name: str = 'Generic Device',
+                 info: Tuple[int, int, int] = (0x03, 0x0001, 0x0001),
+                 rdescs: List[List[int]] = [[
+                        # Generic mouse report descriptor
+                        0x05, 0x01,  # .Usage Page (Generic Desktop)        0
+                        0x09, 0x02,  # .Usage (Mouse)                       2
+                        0xa1, 0x01,  # .Collection (Application)            4
+                        0x09, 0x02,  # ..Usage (Mouse)                      6
+                        0xa1, 0x02,  # ..Collection (Logical)               8
+                        0x09, 0x01,  # ...Usage (Pointer)                   10
+                        0xa1, 0x00,  # ...Collection (Physical)             12
+                        0x05, 0x09,  # ....Usage Page (Button)              14
+                        0x19, 0x01,  # ....Usage Minimum (1)                16
+                        0x29, 0x03,  # ....Usage Maximum (3)                18
+                        0x15, 0x00,  # ....Logical Minimum (0)              20
+                        0x25, 0x01,  # ....Logical Maximum (1)              22
+                        0x75, 0x01,  # ....Report Size (1)                  24
+                        0x95, 0x03,  # ....Report Count (3)                 26
+                        0x81, 0x02,  # ....Input (Data,Var,Abs)             28
+                        0x75, 0x05,  # ....Report Size (5)                  30
+                        0x95, 0x01,  # ....Report Count (1)                 32
+                        0x81, 0x03,  # ....Input (Cnst,Var,Abs)             34
+                        0x05, 0x01,  # ....Usage Page (Generic Desktop)     36
+                        0x09, 0x30,  # ....Usage (X)                        38
+                        0x09, 0x31,  # ....Usage (Y)                        40
+                        0x15, 0x81,  # ....Logical Minimum (-127)           42
+                        0x25, 0x7f,  # ....Logical Maximum (127)            44
+                        0x75, 0x08,  # ....Report Size (8)                  46
+                        0x95, 0x02,  # ....Report Count (2)                 48
+                        0x81, 0x06,  # ....Input (Data,Var,Rel)             50
+                        0xc0,        # ...End Collection                    52
+                        0xc0,        # ..End Collection                     53
+                        0xc0,        # .End Collection                      54
+                     ]]):
```

Comment:
Do we really need to have default values for `info` and `rdescs` here?

I think we better not, so that we are actually forced to add those values in the subclasses or when using this one.

If you really need to have one with default values, just subclass this one as `SimpleMouseDevice`.

Reply:
We don't. We had them in the old class because you asked me to, so I kept them in the rewrite.